### PR TITLE
Sort by X Y

### PR DESCRIPTION
**Pull request by arniebradfo**
I really needed a way to sort artboards by their X Y position, so I added it your plugin. There are two new options:
- Sort Artboards Left to Right, Top to Bottom
- Sort Artboards Top to Bottom, Left to Right

I did modify the plugin to **only sort the selected artboards**, move them to the top, and retain the selection. I thought it was weird to always sort everything? seemed like a bug? This also makes it slightly faster. I also refactored the code a little bit. The meat of the changes are in: [src/commands/sortArtboards.js](https://github.com/rodi01/Artboard-Plus-XD/pull/13/files#diff-3a13df2e4781b982a8355468b6b7eb44)

Anyways. I thought I'd open a PR instead of keeping this to myself.

### CODE REVIEW

**Comment by rodi01** [comment]
You're awesome @arniebradfo! Thanks for helping out with the plugin. 

I'm trying your changes and I'm getting an error: "artboard is not defined"

I think it's related to line 11 of the utils.js file. 
`return (artboard instanceof Artboard)`

File: src/helpers/utils.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 /**
  * Check if node is artboard
  *
  * @param {Object} node
  * @returns
  */
 function isArtboard(node) {
-  return node.constructor.name === "Artboard"
+  // return node.constructor.name === "Artboard"
+  return (artboard instanceof Artboard)
```
Comment (arniebradfo) [inline src/helpers/utils.js:11]: @rodi01 try reverting that line to the line above it?

Comment (arniebradfo) [reply src/helpers/utils.js:11]: woops nevermind...

**Comment by arniebradfo** [comment]
Thought I had fixed that. Honestly don't know how it ran on my computer? 0abe995 should fix it. Try again?

**Comment by rodi01** [comment]
@arniebradfo just to let you know that Adobe rejected the update. Here is their explanation:


"The plugin does not perform the desired action when a certain button is clicked. When user clicks on (Sort Artboard by Name A-Z, Sort Artboard by Name Z-A, Sort Artboard Left to Right, Top to Bottom and Sort Artboard Top to Bottom, Left to Right) then plugin doesn’t sort the artboard."

Maybe it's just the wording of the menu item or it needs some kind of warning. I'll work on it once I have time, if you have any ideas please let me know.

Cheers

**Comment by arniebradfo** [comment]
@rodi01 I tested it again, and it definitely works as expected...

The wording of the response makes me think they misunderstand the desired result? _"Sort Artboard Left to Right"_ actually reads _"Sort Artboard**s** Left to Right"_ (plural artboards, with an s). And _"doesn’t sort **the** artboard"_ sounds like they only selected one artboard.

Makes me think they selected only one artboard and expected the command to sort it alphabetically against the rest of the artboards globally? (Even though that wouldn't be possible and doesn't make sense.) We could add an error when only one artboard is selected, but I think its unnecessary. Could you just request it be retested, and provide some context?

Or maybe they expected the artboards to rearrange their xy-positions on the canvas, not the z-order in the layers panel. 

IDK. this doesn't make sense to me.

**Pull request merged**